src/functional/swing_slide_filter.zig

```diff
@@ -0,0 +1,411 @@
+        // Evaluate the upper and lower bound lines at current timestamp
+        // to obtain the upper and lower limits of the current point.
+        const upper_limit = evaluateLineAtTime(upper_bound_line, current_timestamp);
+        const lower_limit = evaluateLineAtTime(lower_bound_line, current_timestamp);
+
+        if ((upper_limit < (uncompressed_values[current_timestamp] - adjusted_error_bound)) or
+            (lower_limit > (uncompressed_values[current_timestamp] + adjusted_error_bound)))
+        { // Recording mechanism (the current points is outside the limits).
+
+            try appendValue(current_segment.start_value, compressed_values);
+            const segment_size = current_timestamp - current_segment.start_time - 1;
+            if (segment_size > 1) {
+                // Denominator of Eq. (6).
+                const sum_square: f80 = @floatFromInt(
+                    segment_size * (segment_size + 1) * (2 * segment_size + 1) / 6,
+                );
+
+                // Get optimal slope that minimizes the squared error Eq. (5).
+                const slope: f80 = @max(
+                    @min(slope_derivate / sum_square, upper_bound_line.slope),
+                    lower_bound_line.slope,
+                );
+
+                current_line.slope = slope;
+                current_line.intercept = computeInterceptCoefficient(
+                    slope,
+                    current_segment.start_time,
+                    uncompressed_values[current_segment.start_time],
+                );
+                const end_value: f64 = evaluateLineAtTime(current_line, current_timestamp - 1);
+                try appendValue(end_value, compressed_values);
+            } else {
+                try appendValue(current_segment.end_value, compressed_values);
+            }
+
+            try appendIndex(current_timestamp, compressed_values);
+
+            // Update the current segment.
+            current_segment.start_time = current_timestamp;
+            current_segment.start_value = uncompressed_values[current_timestamp];
+
+            // Catch edge case (only one point left).
+            // If `current_timestamp+1 >= uncompressed_values.len`
+            // then `uncompressed_values[current_timestamp + 1]` will return index out of bound error.
+            if (current_timestamp + 1 < uncompressed_values.len) {
+                current_segment.end_time = current_timestamp + 1;
+                current_segment.end_value = uncompressed_values[current_timestamp + 1];
+
+                updateSwingLine(current_segment, &upper_bound_line, adjusted_error_bound);
+                updateSwingLine(current_segment, &lower_bound_line, -adjusted_error_bound);
+
+                current_timestamp += 1;
+                slope_derivate = computeSlopeDerivate(current_segment);
+            } else { // Create line with slope 0 and intercept equal to the current value.
```

Comment:
Maybe make Line 183 only contain the comment for consistency with regards to placement of comments.

Reply:
Fixed.